src/lib/annotations/AnnotationThread.js

```diff
@@ -149,30 +155,24 @@ class AnnotationThread extends EventEmitter {
         // Changing state from pending
         this.state = STATES.hover;
         // Save annotation on server
-        this.annotationService
+        return this.annotationService
             .create(annotationData)
             .then((savedAnnotation) => this.updateTemporaryAnnotation(tempAnnotation, savedAnnotation))
-            .catch(() => {
-                // Remove temporary annotation
-                this.deleteAnnotation(tempAnnotationID, /* useServer */ false);
-
-                // Broadcast error
-                this.emit('annotationcreateerror');
-            });
+            .catch(() => this.handleThreadSaveError(tempAnnotationID));
     }
 
     /**
      * Deletes an annotation.
      *
      * @param {string} annotationID - ID of annotation to delete
      * @param {boolean} [useServer] - Whether or not to delete on server, default true
-     * @return {void}
+     * @return {Promise} - Annotation delete promise
      */
     deleteAnnotation(annotationID, useServer = true) {
         // Ignore if no corresponding annotation exists in thread or user doesn't have permissions
         const annotation = this.annotations.find((annot) => annot.annotationID === annotationID);
         if (!annotation || (annotation.permissions && !annotation.permissions.can_delete)) {
-            return;
+            return Promise.resolve();
```

Comment:
Could this be a Promise.reject() with a message that the annotation either "does not exist" or "cannot be deleted"?

Reply:
Oh. Yea i think that's totally fair. I'm not sure it make sense to reject the promise later on in this method if `useServer = false` but here i think it makes sense. You don't have permission to do what you're trying to do so therefore should get an error